neurom/core/neuron.py

```diff
@@ -156,8 +155,4 @@ def i_neurite(self, iterator_type, mapping=None, tree_filter=None):
             mapping: optional function to apply to the iterator's target.
             tree_filter: optional top level filter on properties of neurite tree objects.
         '''
-        nrt = (self.neurite_trees if tree_filter is None
-               else filter(tree_filter, self.neurite_trees))
-
-        chain_it = chain(*imap(iterator_type, nrt))
-        return chain_it if mapping is None else tr.imap_val(mapping, chain_it)
+        return tr.i_chain(self.neurites, iterator_type, mapping, tree_filter)
```

Comment:
typing tr.i_chain(self.neurites, ...) vs self.i_neurites_chains(....) is about the same, and is more explicit, imo.  No biggie though